### ISSUE TITLE
The Localization is incorret.

### ISSUE THREAD

**Issue by DapengFeng** [colmap/colmap#2737]
**Describe the bug**
The localization process has encountered inaccuracies within the 'Truck' and 'Courthouse' scenarios in the 'Tanks and Temples' project. It is crucial to address these discrepancies to ensure the scenarios are culturally and contextually appropriate for the target audience.

**To Reproduce**
```
for name in Courthouse Truck; do
colmap automatic_reconstructor --image_path ~/data/tanks_and_temples/trainingdata/$name/images  --workspace_path ~/data/tanks_and_temples/trainingdata/$name
done
```

**Expected behavior**

**Screenshots**

![Screenshot from 2024-08-28 09-13-51](https://github.com/user-attachments/assets/1e155963-876d-45bb-9d8e-0d74b8dc7f9f)
![Screenshot from 2024-08-28 09-14-38](https://github.com/user-attachments/assets/2faf5c28-b777-4f41-b2f8-13c1db102d35)

**Environment:**
 - OS: Ubuntu 22.04
 - COLMAP Version:  3.11.0.dev0
 - Capture Device: Tanks and Temples Dataset
 - Ceres-Solver: 2.1.0

**Comment by tsattler** [comment]
I had a quick look into this for the Truck scene. It seems that the features extracted with the `automatic_reconstructor`. At least for me, it extracted the features on the CPU and resulted in two sides of the truck being superimposed.

If I run the pipeline (`feature_extraction` with shared intrinsics, `exhaustive_matcher`, `mapper`) from the command line, the sparse reconstruction looks fine, no super-imposed structure. If I run the pipeline with CPU-extracted features, everything is fine, too. If I run the reconstruction process with the database created by the `automatic_reconstructor` in the gui, I run into the same problem as the `automatic_reconstructor`.

@ahojnnes I guess there is some change in the parameters of the features?

**Comment by ahojnnes** [comment]
How many images are in the scene? The automatic reconstruction switches from exhaustive to vocabulary tree matching at some threshold. The default quality setting may also adjust the options?

**Comment by tsattler** [comment]
It's 251 images.

**Comment by ahojnnes** [comment]
Yes, so if the number of images is more than 200, the automatic reconstruction switches to vocabulary tree matching.

**Comment by ahojnnes** [comment]
Only strange thing is that the command above shows now specification of the vocabulary tree path. @DapengFeng Just to confirm, you are not specifying the path?

**Comment by tsattler** [comment]
This worked for me without any error: `colmap automatic_reconstructor --image_path Truck/ --workspace tmp_models/`

**Comment by DapengFeng** [comment]
> Only strange thing is that the command above shows now specification of the vocabulary tree path. @DapengFeng Just to confirm, you are not specifying the path?

Yeah, I have not specified the path for the vocabulary tree.

**Comment by DapengFeng** [comment]
An unusual observation has been made regarding the `tanks_and_temples` dataset. While most scenes appear to be reconstructed accurately, the application of the `colmap automatic_reconstructor` command on specific scenes, namely `Courthouse` and `Truck`, has resulted in incorrect camera localization.

**Comment by ahojnnes** [comment]
Colmap can easily suffer from incorrect results when dealing with symmetric scenes. It was no built-in way to mitigate them. You may want to look into approaches like Doppelganger, which work quite well.

**Comment by ahojnnes** [comment]
Closing: symmetric scene issue; expected behavior explained by maintainer.
